core-backend/sandbox/src/funcs.rs

```diff
@@ -126,17 +140,21 @@ where
         let (pid_value_ptr, payload_ptr, len, delay, err_mid_ptr) = args.iter().read_5()?;
 
         ctx.run(|ctx| {
+            let read_hash_val = ctx.register_read_as(pid_value_ptr);
+            let read_payload = ctx.register_read(payload_ptr, len);
+            let write_err_mid = ctx.register_write_as(err_mid_ptr);
+
             let HashWithValue {
                 hash: destination,
                 value,
-            } = ctx.read_memory_as(pid_value_ptr)?;
-            let payload = ctx.read_memory(payload_ptr, len)?.try_into()?;
+            } = ctx.read_as(read_hash_val)?;
+            let payload = ctx.read(read_payload)?.try_into()?;
```

Comment:
could you please clarify, whether it's intended to first read to `RuntimeBuffer` and then convert to `Payload`, because the size of the first one is larger than the second one. If we intended to restrict `read` to read bytes only of the `Payload` size, than we have skipped that check here.

Reply:
Discussed I think in DM, isn't it ?